### ISSUE TITLE
Implement an <Icon> element used in Overlays

### ISSUE THREAD

**Issue by cleder** [cleder/fastkml#131]
Basic Implementation in #123 still needs tests

**Comment by cleder** [comment]
# `<Icon>`
https://developers.google.com/kml/documentation/kmlreference#icon
## Syntax
```
<Icon id="ID">
  <!-- specific to Icon -->
  <href>...</href>                         <!-- anyURI -->
  <gx:x>0<gx:x/>                           <!-- int -->
  <gx:y>0<gx:y/>                           <!-- int -->
  <gx:w>-1<gx:w/>                          <!-- int -->
  <gx:h>-1<gx:h/>                          <!-- int -->
  <refreshMode>onChange</refreshMode>
    <!-- kml:refreshModeEnum: onChange, onInterval, or onExpire -->
  <refreshInterval>4</refreshInterval>     <!-- float -->
  <viewRefreshMode>never</viewRefreshMode>
    <!-- kml:viewRefreshModeEnum: never, onStop, onRequest, onRegion -->
  <viewRefreshTime>4</viewRefreshTime>     <!-- float -->
  <viewBoundScale>1</viewBoundScale>       <!-- float -->
  <viewFormat>...</viewFormat>             <!-- string -->
  <httpQuery>...</httpQuery>               <!-- string -->
</Icon>
```
## Description
Defines an image associated with an Icon style or overlay. The required <href> child element defines the location of the image to be used as the overlay or as the icon for the placemark. This location can either be on a local file system or a remote web server. The` <gx:x>`, `<gx:y>`, `<gx:w>`, and` <gx:h>` elements are used to select one icon from an image that contains multiple icons (often referred to as an icon palette.

**Comment by cleder** [comment]
merged in #182 needs more tests
